code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -702,6 +731,27 @@ var/global/list/obj/item/device/pda/PDAs = list()
 			active_conversation = null
 			if(mode==21)
 				mode=2
+		
+		
+		if("Filter") // Filters through available pdas

```

Comment:
Just realized. This entire if structure below this can be simplified to one line:
```
		if("Filter")
			if (href_list["option"])
				pdafilter = sanitize_integer(text2num(href_list["option"]), 0, 8, pdafilter)
```

This means you change the helper links to have the following look:
`{{:helper.link( 'All',data.pdafilter == 0? 'null' : close, {'choice' : "Filter", 'option' : 0}, null, 'fixedLeft')}}`

And for ref, sanitize integer function has these params:
`/proc/sanitize_integer(number, min=0, max=1, default=0)`